src/TodoItem.js

```diff
@@ -0,0 +1,165 @@
+import { useState } from "react";
+import { styled } from "styled-components";
+
+const TodoItem = ({ onEdit, onDelete, moveItem, value, id, isDone }) => {
+  const [isEdit, setIsEdit] = useState(false);
+  const [localContent, setLocalContent] = useState(value); //수정된 데이터
+
+  const toggleIsEdit = (e) => {
+    e.stopPropagation();
+    setIsEdit(!isEdit);
+  };
+
+  //수정 취소
+  const handleQuitEdit = (e) => {
+    e.stopPropagation();
+    setIsEdit(false);
+    setLocalContent(value);
+  };
+
+  //수정 완료
+  const handleCompleteEdit = (e) => {
+    e.stopPropagation();
+    if (localContent.trim() == "") {
+      return;
+    }
+    if (window.confirm(`할 일을 수정하시겠습니까?`)) {
+      onEdit(id, localContent);
+      toggleIsEdit(e);
+    }
+  };
+
+  return (
+    <DataContainer
+      className="TodoItem"
+      onClick={() => {
+        moveItem(id);
+      }}
+    >
+      <Datalist>
+        {isEdit ? (
+          <>
+            <EditTextarea
+              value={localContent}
+              onClick={(e) => {
+                e.stopPropagation();
+              }}
+              onChange={(e) => setLocalContent(e.target.value)}
+            />
+          </>
+        ) : (
+          <>{value}</>
+        )}
+      </Datalist>
+      <ButtonsContainer>
+        {isEdit ? (
+          <>
+            <EditingBtn
+              onClick={(e) => {
+                handleQuitEdit(e);
+              }}
```

Comment:
만약 함수의 매개변수가 (event) 하나뿐이라면 arrow Function을 생략할 수 있습니다
```suggestion
              onClick={handleQuitEdit}
```